app/services/auth.py

```diff
@@ -0,0 +1,100 @@
+from configparser import ConfigParser
+import os
+from typing import Optional
+
+import jwt
+from fastapi import Depends, HTTPException, status
+from fastapi.security import SecurityScopes, HTTPAuthorizationCredentials, HTTPBearer
+
+def get_auth0_config():
+    """Sets up configuration for the app"""
+
+    env = os.getenv("ENV", ".config")
+
+    # Read config from .config file or environment variables
+    if env == ".config":
+        config = ConfigParser()
+        config.read(".config")
+        config = config["AUTH0"]
+    else:
+        config = {
+            "DOMAIN": os.getenv("AUTH0_DOMAIN"),
+            "API_AUDIENCE": os.getenv("AUTH0_API_AUDIENCE"),
+            "ISSUER": os.getenv("AUTH0_ISSUER"),
+            "ALGORITHMS": os.getenv("AUTH0_ALGORITHMS"),
+        }
+    for key, value in config.items():
+        if value == "":
+            print(f"Missing config: {key}")
+            exit(1)
+    return config
+
+class UnauthorizedException(HTTPException):
+    def __init__(self, detail: str, **kwargs):
+        """Returns HTTP 403"""
+        super().__init__(status.HTTP_403_FORBIDDEN, detail=detail)
+
+
+class UnauthenticatedException(HTTPException):
+    def __init__(self):
+        super().__init__(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Requires authentication"
+        )
```

Comment:
are we sure this is the correct error mapping? shouldn't UnauthorizedException return an unaurthorized status?

Reply:
WDYM by unauthorized status?
403 is for when you're logged in but not allowed to do something which afaik is the same as unauthorized
I took this code from auth0's example

Reply:
I mean, 
UnauthorizedException-> HTTP_403_FORBIDDEN
UnauthenticatedException -> HTTP_401_UNAUTHORIZED

look unintuitive but don't care really